packages/nimble-components/src/menu-button/tests/menu-button-matrix.stories.ts

```diff
@@ -35,39 +40,72 @@ const partVisibilityStates = [
 ] as const;
 type PartVisibilityState = (typeof partVisibilityStates)[number];
 
+const openStates = [
+    ['', false],
+    ['Open', true]
+] as const;
+type OpenState = (typeof openStates)[number];
+
 const appearanceStates = Object.entries(ButtonAppearance).map(
     ([key, value]) => [pascalCase(key), value]
 );
 type AppearanceState = (typeof appearanceStates)[number];
 
 // prettier-ignore
 const component = (
-    [iconVisible, labelVisible, endIconVisible]: PartVisibilityState,
+    [interactionName, interaction]: InteractionState,
     [disabledName, disabled]: DisabledState,
+    [iconVisible, labelVisible, endIconVisible]: PartVisibilityState,
+    [openName, open]: OpenState,
     [appearanceName, appearance]: AppearanceState
 ): ViewTemplate => html`
     <${menuButtonTag}
+        class="${() => interaction}"
         appearance="${() => appearance}"
+        ?open="${() => open}"
         ?disabled=${() => disabled}
         ?content-hidden=${() => !labelVisible}
         style="margin-right: 8px; margin-bottom: 8px;">
             ${when(() => iconVisible, html`<${iconKeyTag} slot="start"></${iconKeyTag}>`)}
-            ${() => `${appearanceName!} Menu Button ${disabledName}`}
+            ${() => `${interactionName} ${openName} ${appearanceName!} Menu Button ${disabledName}`}
             ${when(() => endIconVisible, html`<${iconArrowExpanderDownTag} slot="end"></${iconArrowExpanderDownTag}>`)}
-
-        <${menuTag} slot="menu">
-            <${menuItemTag}>Item 1</${menuItemTag}>
-            <${menuItemTag}>Item 2</${menuItemTag}>
-        </${menuTag}>
     </${menuButtonTag}>
 `;
 
 export const menuButtonThemeMatrix: StoryFn = createMatrixThemeStory(
-    createMatrix(component, [
-        partVisibilityStates,
-        disabledStates,
-        appearanceStates
-    ])
+    createMatrix(
+        component,
+        [
+            nonInteractionStates,
+            disabledStates,
+            partVisibilityStates,
+            openStates,
+            appearanceStates
+        ],
+        // Disabled and open is not a valid state
+        (
+            _interactionState: InteractionState,
+            disabledState: DisabledState,
+            _partVisibilityState: PartVisibilityState,
+            openState: OpenState
+        ) => {
+            return disabledState[0] !== 'Disabled' || openState[0] !== 'Open';
```

Comment:
I don't think I like this filter feature concept.. Our goal is that all states a control can be put in should behave sanely and consistently (i.e. minimally tested via matrix tests), even if when set they are not fully supported by usage guidance. The behavior should be well-known even if not recommended. This filter concept on the matrix tests encourages the opposite, putting gaps in our testing. We should discuss this outside of PR comments, i.e. post stand-up topic, etc. FYI @jattasNI

Reply:
From our discussion in the team meeting, it sounds like the two main points are:
- the main matrix test for a component should not exclude specific combinations, because we need to ensure that no configuration causes things to blow up
- excluding certain configurations from interaction matrix tests is fine, but using a filter function makes it harder to understand exactly what is being excluded

I introduced the filter functionality because, for interactions on disabled controls, I only wanted to test the hover behavior. I.e. there isn't a way for a disabled control to be active and/or focused. I can think of a few directions to go in:
1. Use the filter behavior, like I have in this PR.
2. Break the one interaction test into two: one for disabled, and one for enabled. I don't like that for two reasons: (1) it increases the number of snapshots, and (2) conceptually, it seems like an arbitrary split.
3. Test all interaction states of disabled controls, even if they don't make sense. I dislike this because it makes the snapshot noisier to read.
4. Collapse the `interactionStates` and `disabledStates` matrix dimensions into one, with values like: `enabled`, `enabled-hover`, `enabled-hover-active`, `enabled-active`, `enabled-focus`, `disabled`, and `disabled-hover`. I don't like this idea.

My vote would be to stick with option 1, but I'm open to other ideas or arguments.

Reply:
I'm working on a 5th alternative which is to break-up matrix into smaller, targeted pieces that compose together but it's still WIP but FYI I'm pushing to this branch directly

Reply:
@m-akinc updated the PR with my proposed approach. Has the following characteristics:
- Interaction states are currently like background colors. Most controls are not going to need control specific integration so made a helper similar to the theme matrix helper.
- Switched from filtering of states to opting into states for each mode of interactions. I think this approach is easier to follow as the states used are explicit. I don't have to jump back and forth between the set of states and mentally running the filter to know what to expect. The states are all listed explicitly.
- There no longer is ordering dependence across dimensions, i.e. the interaction and disabled dimensions don't have to be first and second. Allows for controls to place things in an order that makes sense for the control. For example the menu-button making open states the first dimension.

Let me know if you have any questions or concerns. If not I'll do a final pass before approve.

Reply:
Took another pass through and feel pretty confident with the pattern so resolving the comment. But if want to discuss feel free to re-open 👍